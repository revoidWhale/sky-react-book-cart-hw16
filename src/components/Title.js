/* eslint-disable react/jsx-props-no-spreading */
import styled from 'styled-components'

const StyledTitle = styled.h1`
  color: ${(props) => props.color || 'green'};
`

const Title = (props) => <StyledTitle {...props} />

export default Title
