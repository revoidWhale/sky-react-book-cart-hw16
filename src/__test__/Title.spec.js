import Title from '../hw/Title'
import Enzyme, { render } from 'enzyme'
import Adapter from '@wojtekmaj/enzyme-adapter-react-17'
import React from 'react'

Enzyme.configure({ adapter: new Adapter() })

describe('Title component', () => {
  it('should render Title component with props', () => {
    const view = render(<Title title="Test title" />)
    expect(view).toMatchSnapshot()
  })

  it('should render Title component without props', () => {
    const view = render(<Title />)
    expect(view).toMatchSnapshot()
  })
})
