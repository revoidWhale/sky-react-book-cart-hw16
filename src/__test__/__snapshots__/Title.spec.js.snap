// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title component should render Title component with props 1`] = `
LoadedCheerio {
  "0": <h1
    class="title"
  >
    Test title
  </h1>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;

exports[`Title component should render Title component without props 1`] = `
LoadedCheerio {
  "0": <h1
    class="title"
  >
    Simple title
  </h1>,
  "_root": LoadedCheerio {
    "0": Document {
      "children": Array [
        <html>
          <head />
          <body />
        </html>,
      ],
      "endIndex": null,
      "next": null,
      "parent": null,
      "prev": null,
      "startIndex": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
